### PR TITLE
Only add -lrt in cases where we definitely need it and adding it is safe...

### PR DESCRIPTION
**Pull request by gbtitus**
....

### CODE REVIEW

**Comment by gbtitus** [comment]
@thomasvandoren can you take a look?

File: util/chplenv/chpl_3p_qthreads_configs.py [hunk 1]
```diff
@@ -11,9 +11,13 @@ def get_uniq_cfg_path():
...
-                                                         'libqthread.la',
-                                                         '-lrt'])
+    link_args = \
+        third_party_utils.default_get_link_args('qthread',
+                                                ucp=get_uniq_cfg_path(),
+                                                libs=['libqthread_chpl.la',
+                                                      '-lchpl',
+                                                      'libqthread.la'])
+    if ( chpl_compiler.get('target').startswith('cray-prgenv') and
+         chpl_llvm.get() == 'llvm' ):
```
Comment (thomasvandoren) [inline util/chplenv/chpl_3p_qthreads_configs.py]: Do you think `chpl_llvm.get() != 'none'` is more future proof, e.g. if we ever support CHPL_LLVM=system?. No need to change it now, I am just curious.

**Comment by thomasvandoren** [comment]
Looks good :+1:

File: util/chplenv/chpl_3p_qthreads_configs.py (hunk 1 above)
Comment (gbtitus) [inline util/chplenv/chpl_3p_qthreads_configs.py]: Ah, actually that is a good point and I will change it before merging. Thanks!

**Pull request merged**